hudi-spark-datasource/hudi-spark/src/test/scala/org/apache/hudi/functional/TestPayloadDeprecationFlow.scala

```diff
@@ -78,27 +95,41 @@ class TestPayloadDeprecationFlow extends SparkClientFunctionalTestHarness {
     val secondUpdate = spark.createDataFrame(secondUpdateData).toDF(columns: _*)
     secondUpdate.write.format("hudi").
       option(OPERATION.key(), "upsert").
-      option(HoodieCompactionConfig.INLINE_COMPACT.key(), "true").
+      option(HoodieCompactionConfig.INLINE_COMPACT.key(), "false").
       option(HoodieCompactionConfig.INLINE_COMPACT_NUM_DELTA_COMMITS.key(), "1").
-      options(opts).
       mode(SaveMode.Append).
       save(basePath)
-    // 4. Validate.
-    val df = spark.read.format("hudi").options(opts).load(basePath)
+    // 4. Add a trivial update to trigger payload class mismatch.
+    val thirdUpdateData = Seq(
+      (12, "3", "rider-CC", "driver-CC", 33.90, "i"))
+    val thirdUpdate = spark.createDataFrame(thirdUpdateData).toDF(columns: _*)
+    if (!payloadClazz.equals(classOf[MySqlDebeziumAvroPayload].getName)) {
```

Comment:
why do we need to test this in this test? 
why not add a diff simple test just to test updating a diff payload class? we do not need to test this one for all diff payload classes. just doing it for 1 combination should suffice

Reply:
not really much of a concern. punting this for now.